homeassistant/core.py

```diff
@@ -199,6 +196,13 @@ def is_callback(func: Callable[..., Any]) -> bool:
     return getattr(func, "_hass_callback", False) is True
 
 
+class _Hass(threading.local):
+    hass: HomeAssistant | None = None
```

Comment:
We should move this down to just before the definition of the `HomeAssistant` class I think.
Also, add a docstring:

```suggestion
class _Hass(threading.local):
   """Container which makes a HomeAssistant instance available to the event loop."""
    hass: HomeAssistant | None = None
```